src/Tribe/Admin_List.php

```diff
@@ -155,6 +160,42 @@ public static function sort_by_event_date( Array $clauses, WP_Query $wp_query )
 			return $clauses;
 		}
 
+		/**
+		 * Defines custom logic for filtering events table by aggregator record.
+		 *
+		 * @param array<string> $clauses    SQL clauses for fetching posts.
+		 * @param WP_Query      $wp_query   A paginated query for items.
+		 *
+		 * @return array<string>            Modified SQL clauses.
+		 */
+		public static function filter_by_aggregator_record( array $clauses, WP_Query $wp_query ) {
+			// Check for event post type.
+			if ( $wp_query->get( 'post_type' ) !== TEC::POSTTYPE ) {
+				return $clauses;
+			}
+
+			// Check if filtering by aggregator record.
+			$record_num = (int) tribe_get_request_var( 'aggregator_record', 0 );
+			if ( 0 === $record_num ) {
+				return $clauses;
+			}
+
+			global $wpdb;
+			
+			$table_alias = 'ea_record_' . substr( uniqid( 'ea_record', true ), 0, 10 );
+			// Add the record meta query if it is missing.
+			if ( ! preg_match( '/\\stribe_aggregator_record\\s/', $clauses['join'] ) ) {
```

Comment:
Since the table alias does not have any trace of the `tribe_aggregator_record` string in it, this will always be `true` and will make the code buggy.

I have outlined two approaches in my suggestion:
1. to have a less specific name for the table alias, **possibly** random.
2. to add spaces before and after the table alias to pick that up.

Committing both suggestions to code does not result in correctly working code: the filtering will always be applied.

Reply:
Good catch. I changed the check to include the new table alias and kept the spaces in place.